gax-java/gax/src/main/java/com/google/api/gax/rpc/EndpointContext.java

```diff
@@ -320,6 +330,7 @@ String mtlsEndpointResolver(
     public EndpointContext build() throws IOException {
       // The Universe Domain is used to resolve the Endpoint. It should be resolved first
       setResolvedUniverseDomain(determineUniverseDomain());
+      setCustomEndpoint(determineCustomEndpoint());
```

Comment:
I'll need to play around with this locally to confirm, but my thoughts as of now is that I'd rather not expose both custom endpoint and resolved endpoint as I think that would be confusing.

If possible, I would prefer something to expose something like a `canUseS2A` flag here. It would check everything that is being checked above in the GrpcChannelProvider logic:
1.  Check env var
2. Check if user supplied a custom endpoint
3. Check if mtls endpoint is set
4. ...

I think everything that is needed for the checks here exists inside the endpointContext.

Reply:
rather not expose both custom endpoint and resolved endpoint as I think that would be confusing
> Agreed on this, and that replacing the usage of customEndpoint/endpointOverride with canUseS2A would be cleaner.

Tried this locally (have not pushed a commit yet) and some points:
- to check the env var in the GrpcChannelProvider impl, we use an `EnvironmentProvider`. I guess in `EndpointContext` we could use System.getenv, but this goes against the pattern of getting other env variables (GOOGLE_CLOUD_ENABLE_DIRECT_PATH_XDS, GOOGLE_API_USE_CLIENT_CERTIFICATE etc.)
- Because of above, my canUseS2A checks everything besides the env var. In the GrpcChannelProvider, I check canUseS2A && isEnvVarSet.

With this change, we set canUseS2A and mtlsEndpoint in the GrpcChannelProvider.

Reply:
@lqiu96  In https://github.com/googleapis/sdk-platform-java/commit/dc4b61e5dfacb93a80e827a245c45d83d1b87707, I implemented changes to use the endpointContext + moved logic to EndpointContext.